tests/app/UnitTests/GitUI.Tests/Editor/Diff/DiffHighlightServiceTests.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Castle.Core;
+using FluentAssertions;
+using GitUI.Editor.Diff;
+using ICSharpCode.TextEditor.Document;
+
+namespace GitUITests.Editor.Diff;
+
+[TestFixture]
+public class DiffHighlightServiceTests
+{
+    [Test]
+    public void GetDifferenceMarkers()
+    {
+        // LineSegment is hard to create. Use TextMarker as implementation type of ISegment for this test.
+        const TextMarkerType dontCare = TextMarkerType.SolidBlock;
+
+        const string identicalPartBefore = "identical_part_before_";
+        const string identicalPartAfter = "_identical_part_after";
+        const string differentRemoved = "RemovedX";
+        const string differentAdded = "AddedY";
+        const string removedLineText = $"-{identicalPartBefore}{differentRemoved}{identicalPartAfter}";
+        const string addedLineText = $"+{identicalPartBefore}{differentAdded}{identicalPartAfter}";
+        const string text = $"{removedLineText}\n{addedLineText}";
+        TextMarker removedLine = new(offset: text.IndexOf(removedLineText), removedLineText.Length, textMarkerType: dontCare);
+        TextMarker addedLine = new(offset: text.IndexOf(addedLineText), addedLineText.Length, textMarkerType: dontCare);
+        const int lineStartOffset = 1;
+
+        IEnumerable<TextMarker> markers = DiffHighlightService.GetDifferenceMarkers(GetCharAt, removedLine, addedLine, lineStartOffset);
+
+        TextMarker[] expectedMarkers =
+        [
+            new TextMarker(offset: addedLine.Offset + lineStartOffset, length: identicalPartBefore.Length, TextMarkerType.SolidBlock),
+            new TextMarker(offset: addedLine.Offset + lineStartOffset + identicalPartBefore.Length + differentAdded.Length, length: identicalPartAfter.Length, TextMarkerType.SolidBlock),
```

Comment:
```suggestion
            new TextMarker(offset: addedLine.Offset + lineStartOffset + identicalPartBefore.Length + differentAdded.Length, length: identicalPartAfter.Length, dontCare),
```

Reply:
`dontCare` does not apply here.
The `MarkerType` of the results should be verified. The improved version in the next PR will not work without it.